snuba/consumers/consumer.py

```diff
@@ -795,31 +814,62 @@ def __init__(
         ):
             self.__process_message_fn = process_message_multistorage_identical_storages
 
-        self.__inner_factory = ConsumerStrategyFactory(
-            prefilter=None,
-            process_message=self.__process_message_fn,
-            collector=partial(
-                build_collector, self.__metrics, self.__storages, commit_log_config
-            ),
-            max_batch_size=max_batch_size,
-            max_batch_time=max_batch_time,
-            processes=processes,
-            input_block_size=input_block_size,
-            output_block_size=output_block_size,
-            initialize_parallel_transform=initialize_parallel_transform,
-            parallel_collect=parallel_collect,
-            dead_letter_queue_policy_creator=self.__dead_letter_policy_creator,
+        self.__collector = partial(
+            build_collector, self.__metrics, self.__storages, commit_log_config
         )
 
     def create_with_partitions(
         self,
         commit: Commit,
         partitions: Mapping[Partition, int],
     ) -> ProcessingStrategy[KafkaPayload]:
+
+        collect = (
+            ParallelCollectStep(
+                self.__collector,
+                CommitOffsets(commit),
+                self.__max_batch_size,
+                self.__max_batch_time,
+                self.__parallel_collect_timeout,
+            )
+            if self.__parallel_collect
+            else CollectStep(
+                self.__collector,
+                CommitOffsets(commit),
+                self.__max_batch_size,
+                self.__max_batch_time,
+            )
+        )
+
+        transform_function = self.__process_message_fn
+
+        inner_strategy: ProcessingStep[MultistorageKafkaPayload]
```

Comment:
Do you actually need to declare this here ?

Reply:
Yeah, i think it was needed. Since it can be either TransformStep or ParallelTransformStep, it doesn't infer the right type.